lib/super_tooltip.dart

```diff
@@ -1,908 +1,1084 @@
-  /// [top], [right], [bottom], [left] position the Tooltip absolute relative to the whole screen
-  double top, right, bottom, left;
-  ///
-  /// A Tooltip can have none, an inside or an outside close icon
-  final ShowCloseButton showCloseButton;
-  ///
-  /// [hasShadow] defines if the tooltip should have a shadow
-  final bool hasShadow;
-  ///
-  /// the stroke width of the border
-  final double borderWidth;
-  ///
-  /// The corder radii of the border
-  final double borderRadius;
-  ///
-  /// The color of the border
-  final Color borderColor;
-  /// 
-  /// The color of the close icon
-  final Color closeButtonColor;
-  /// 
-  /// The size of the close button 
-  final double closeButtonSize;
-  ///
-  /// The length of the Arrow
-  final double arrowLength;
-  /// 
-  /// The width of the arrow at its base
-  final double arrowBaseWidth;
-  ///
-  /// The distance of the tip of the arrow's tip to the center of the target
-  final double arrowTipDistance;
-  ///
-  /// The backgroundcolor of the Tooltip
-  final Color backgroundColor;
-  /// The color of the rest of the overlay surrounding the Tooltip.
-  /// typically a translucent color.
-  final Color outsideBackgroundColor;
-  ///
-  /// By default touching the surrounding of the Tooltip closes the tooltip.
-  /// you can define a rectangle area where the background is completely transparent 
-  /// and the widgets below react to touch
-  final Rect touchThrougArea; 
-  ///
-  /// The shape of the [touchThrougArea]. 
-  final ClipAreaShape touchThroughAreaShape;
-  ///
-  /// If [touchThroughAreaShape] is [ClipAreaShape.rectangle] you can define a border radius
-  final double touchThroughAreaCornerRadius;
-  ///
-  /// Let's you pass a key to the Tooltips cotainer for UI Testing
-  final Key tooltipContainerKey;
-
-  Offset _targetCenter;
-  OverlayEntry _backGroundOverlay;
-  OverlayEntry _ballonOverlay;
-
-  SuperTooltip({
-    this.tooltipContainerKey,
-    @required this.content, // The contents of the tooltip.
-    @required this.popupDirection,
-    this.onClose,
-    this.minWidth,
-    this.minHeight,
-    this.maxWidth,
-    this.maxHeight,
+  Future<void> hideTooltip() {
+    event = _Event.hide;
+    _completer = Completer();
+    notifyListeners();
+    return _completer.future.then((_) => _isVisible = false);
+  }
+
+  void complete() {
+    if (!_completer.isCompleted) {
+      _completer.complete();
+    }
+  }
+}
+
+// TODO: Change preferredDirectionn direction and hot reload and assertion error
+
+class SuperTooltip extends StatefulWidget {
+  const SuperTooltip({
+    Key key,
+    @required this.content,
+    this.preferredDirection = PreferredDirection.down,
+    this.controller,
+    this.onLongPress,
+    this.onShow,
+    this.onHide,
+    this.snapsFarAwayVertically = false,
+    this.snapsFarAwayHorizontally = false,
     this.top,
     this.right,
     this.bottom,
     this.left,
-    this.minimumOutSidePadding = 20.0,

```

Comment:
Why did you remove them?

Reply:
The word padding didn't sound appropriate to me, as it implied the spacing inside of the widget as opposed to a margin, which is spacing outside the widget. Do you agree? As for the `minimumOutside` prior to the word, sorry, I just forgot to add that. Will add it back in next commit

Reply:
yeah you are right margin is a better term although technically its a padding to the wrapping container.